src/implementation/Server/HTTPServer/actor.ts

```diff
@@ -98,13 +99,18 @@ export default class HTTPServerActor implements IServerActor {
     // @todo: reentrancy id? (https://github.com/dapr/python-sdk/blob/master/ext/flask_dapr/flask_dapr/actor.py#L91)
 
     const dataSerialized = this.serializer.serialize(body);
-    const result = await ActorRuntime.getInstance(this.client.getDaprClient()).invoke(
-      actorTypeName,
-      actorId,
-      methodName,
-      dataSerialized,
-    );
-    return this.handleResult(res, result);
+    try {
+      const result = await ActorRuntime.getInstance(this.client.getDaprClient()).invoke(
+        actorTypeName,
+        actorId,
+        methodName,
+        dataSerialized,
+      );
+      return this.handleResult(res, result);
+    } catch (err) {
+      res.statusCode = HttpStatusCode.BAD_REQUEST;
```

Comment:
Can `invoke` fail for any other thing than non-existent method scenario? Either we can make this a general error, like `HttpStatusCode.INTERNAL_SERVER_ERROR`, or we should only catch errors where a certain string is matched and then return `BAD_REQUEST` for it.

Reply:
Yup, it can fail for any other reason as well. I have changed it to INTERNAL_SERVER_ERROR.